tests/xtimer_now_irq/main.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2019 Freie Universität Berlin,
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       testing xtimer_now_usec function with irq disabled
+ *
+ *
+ * @author      Julian Holzwarth <julian.holzwarth@fu-berlin.de>
+ *
+ */
+
+#include <stdio.h>
+#include "xtimer.h"
+#include "irq.h"
+#include "test_utils/interactive_sync.h"
+
+#define TEST_COUNT 4
+
+int main(void)
+{
+    test_utils_interactive_sync();
+    puts("xtimer_now_irq test application.\n");
+
+    while (TEST_COUNT) {
+        unsigned int state = irq_disable();
+        uint32_t t1 = xtimer_now_usec();
+        xtimer_spin(xtimer_ticks_from_usec((uint32_t)(~XTIMER_MASK) / 2));
```

Comment:
Why do you use `XTIMER_MASK` ? Wouldn't it be simpler to directly use a usec value ? This way it would be defined globally and an approaching value could be used in the Python test script.

Reply:
Actually the value handed to spin should be ticks, so interpreting the XTIMER_MASK as usecs doesn't make sense. But using XTIMER_MASK so the value is relative to the maximum value of the timer is good to select the value so the overflow can actually be assumed to happen while interrupts are disabled.

Reply:
This should be addressed with the new commit. Correct?

Reply:
@MichelRottleuthner